lte/gateway/python/integ_tests/Makefile

```diff
@@ -70,11 +70,32 @@ precommit: prepare_environment
 	-$(foreach test,$(PRECOMMIT_TESTS),$(call execute_test,$(test));)
 	$(call apply_final_status)
 
-.PHONY: integ_test
-integ_test: prepare_environment
-	-$(foreach test,$(EXTENDED_TESTS) $(PRECOMMIT_TESTS),$(call execute_test,$(test));)
+.PHONY: precommit_tests_noncontainer
+precommit_tests_noncontainer: prepare_environment
+	-$(foreach test,$(PRECOMMIT_TESTS_NOT_CONTAINER),$(call execute_test,$(test));)
+	$(call apply_final_status)
+
+.PHONY: extended_tests
+extended_tests: prepare_environment
+	-$(foreach test,$(EXTENDED_TESTS),$(call execute_test,$(test));)
+	$(call apply_final_status)
+
+.PHONY: extended_tests_noncontainer
+extended_tests_noncontainer: prepare_environment
+	-$(foreach test,$(EXTENDED_TESTS_NOT_CONTAINER),$(call execute_test,$(test));)
 	$(call apply_final_status)
 
+.PHONY: extended_tests_long
+extended_tests_long: prepare_environment
+	-$(foreach test,$(EXTENDED_TESTS_LONG),$(call execute_test,$(test));)
+	$(call apply_final_status)
+
+.PHONY: integ_test
+integ_test: precommit precommit_tests_noncontainer extended_tests extended_tests_noncontainer extended_tests_long
+
```

Comment:
With this we call `prepare_environment` five times during execution. May this become a performance issue at some point?

Reply:
Consider the following small example:
```makefile
.PHONY: prep
prep:
	echo "prep"

.PHONY: first
first: prep
	echo "first"

.PHONY: blub
second: prep first
	echo "second"

.PHONY: diamond
diamond: first second
	echo "diamond"
```
For me this results in
```bash
$ make diamond
echo "prep"
prep
echo "first"
first
echo "second"
second
echo "diamond"
diamond
```
So I trust make does work correctly with dependency trees, and does not execute this multiple times, when started in the same thread.

On the other hand the dependency definition is logically required, if those targets should be independently usable. And I think they should to be valid targets.